static/app/utils/performance/segmentExplorer/tagKeyHistogramQuery.tsx

```diff
@@ -34,13 +34,15 @@ type ChildrenProps = Omit<GenericChildrenProps<TableData>, 'tableData'> & {
 type QueryProps = DiscoverQueryProps & {
   aggregateColumn: string;
   tagKey: string;
+  tagKeyLimit: number;
   sort?: string | string[];
   children: (props: ChildrenProps) => React.ReactNode;
 };
 
 type FacetQuery = LocationQuery &
   EventQuery & {
     tagKey?: string;
+    tagKeyLimit?: number;
```

Comment:
nit: Would we be able to make `tagKeyLimit` non optional? Seems like `getRequestFunction` is the only function that references `FacetQuery` and some of these props are optional because they're being set on `apiPayload` after it initialization. Might be able to get around this using spread:
```
    const apiPayload: FacetQuery = {
      ...eventView.getEventsAPIPayload(props.location),
      ...{
        aggregateColumn,
        sort: _props.sort ? _props.sort : '-sumdelta',
        tagKey: _props.tagKey,
        tagKeyLimit: _props.tagKeyLimit,
      },
    };
```
You might have a better way to do this lol

Reply:
I'm probably just going to leave it until the follow up backend PR.